controller/node_controller.go

```diff
@@ -403,6 +407,25 @@ func (nc *NodeController) syncNode(key string) (err error) {
 			return err
 		}
 	} else {
+		// check if the role of the node got changed to non-worker
+		kubeNodeLabels := kubeNode.GetLabels()
+		if len(kubeNodeLabels) != 0 {
```

Comment:
If there is no label, we would have a problem too?

Also, what if the label exists but the value is false?

Reply:
Yes. If we don't have a label, then integration test is failing as the test nodes that they create wouldn't have label in them.

I haven't seen any instance in my testing where the label is present and the value is false. Is this a possibility too?

Reply:
I think if we cannot find labels, we should print out an error at least. It's very unlikely to happen but we don't want to fail silently. 

About the false value, it's possible since it's totally implementation dependent. It's better if we check for it too.